core/src/main/scala/org/apache/gearpump/jarstore/FileDirective.scala

```diff
@@ -67,25 +69,32 @@ object FileDirective {
   type FormField = Either[FileInfo, String]
 
   /**
-   * directive to uploadFile, it store the uploaded files
-   * to temporary directory, and return a Map from form field name
+   * Store the uploaded files to temporary directory, and return a Map from form field name
    * to FileInfo.
    */
   def uploadFile: Directive1[Form] = {
-    uploadFileTo(null)
+    Directive[Tuple1[Form]] { inner =>
+      extractMaterializer {implicit mat =>
+        extractExecutionContext {implicit ec =>
+          uploadFileImpl(mat, ec) { filesFuture =>
+            ctx => {
+              filesFuture.map(map => inner(Tuple1(map))).flatMap(route => route(ctx))
```

Comment:
Can we improve code readability here ? It's hard to comprehend what it is doing.